### ISSUE TITLE
VerneMQ failed to start

### ISSUE THREAD

**Issue by conni-mpd** [vernemq/vernemq#1777]
### Environment

- VerneMQ Version: 1.11.0
- OS: Ubuntu 18.04.4 LTS (Bionic)
- Erlang/OTP version: 22 
- VerneMQ configuration (vernemq.conf) or the changes from the default
   - vernemq.conf
`listener.tcp.default = 0.0.0.0:8009`
`log.console.level = debug`
   - vernemq.service 
`[Unit]`
`Description=Runner for VerneMQ`
`After=network.target`
`[Service]`
`WorkingDirectory=/usr/lib/vernemq`
`ExecStart=/usr/sbin/vernemq start`
`ExecStop=/usr/sbin/vernemq stop`
`User=vernemq`
`Type=forking`
`Restart=always`
`RestartSec=45`
`LimitNOFILE=300000`
`[Install]`
`WantedBy=multi-user.target`

- Cluster size/standalone: 1

### Expected behaviour

Expect to start normal. 

### Actual behaviour

After cleaning the syslog for disk space, verneMQ stops working. Rebooting and reinstalling didn't help. Not sure if this is VerneMQ issue or database related issue.  Thanks for helping. 

When checking the broker status:  systemctl status vernemq.service
● vernemq.service - Runner for VerneMQ
   Loaded: loaded (/etc/systemd/system/vernemq.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Mon 2021-04-12 19:02:13 GMT; 44s ago
  Process: 21728 ExecStop=/usr/bin/vernemq stop (code=exited, status=1/FAILURE)
  Process: 21033 ExecStart=/usr/bin/vernemq start (code=exited, status=0/SUCCESS)
 Main PID: 17372 (code=exited, status=1/FAILURE)

Apr 12 19:02:13 vernemq[21728]: Node is not running!
Apr 12 19:02:13 systemd[1]: vernemq.service: Control process exited, code=exited status=1
Apr 12 19:02:13 systemd[1]: vernemq.service: Failed with result 'exit-code'.
Apr 12 19:02:13 systemd[1]: Failed to start Runner for VerneMQ.

In console.log: 
2021-04-12 18:45:12.786 [error] <0.202.0> CRASH REPORT Process <0.202.0> with 0 neighbours exited with reason: {shutdown,{gen_server,call,[vmq_plugin_mgr,{enable_system_plugin,vmq_generic_msg_store,[internal]},infinity]}} in application_master:init/4 line 138
2021-04-12 18:45:12.795 [info] <0.232.0>@vmq_storage_engine_leveldb:open_db:145 Opening LevelDB database at "/var/lib/vernemq/msgstore/7"
2021-04-12 18:45:12.907 [info] <0.231.0>@vmq_generic_msg_store:handle_info:246 indexed 1230 offline messages in msg store instance 6
2021-04-12 18:45:12.932 [debug] <0.233.0>@vmq_storage_engine_leveldb:init_state:122 datadir /var/lib/vernemq/msgstore/8 options for LevelDB: [{open,[{block_cache_threshold,33554432},{block_restart_interval,16},{block_size_steps,16},{compression,true},{create_if_missing,true},{delete_threshold,1000},{eleveldb_threads,71},{fadvise_willneed,false},{limited_developer_mem,false},{sst_block_size,4096},{tiered_slow_level,0},{total_leveldb_mem_percent,70},{use_bloomfilter,true},{write_buffer_size,39191303}]},{read,[{verify_checksums,true}]},{write,[{sync,false}]},{fold,[{verify_checksums,true},{fill_cache,false}]}]
2021-04-12 18:45:13.241 [error] <0.167.0> gen_server disksup terminated with reason: {port_died,normal}
2021-04-12 18:45:13.241 [error] <0.167.0> CRASH REPORT Process disksup with 0 neighbours exited with reason: {port_died,normal} in gen_server:handle_common_reply/8 line 751
2021-04-12 18:45:13.241 [error] <0.168.0> gen_server memsup terminated with reason: {port_died,normal}
2021-04-12 18:45:13.267 [error] <0.166.0> Supervisor os_mon_sup had child disksup started with disksup:start_link() at <0.167.0> exit with reason {port_died,normal} in context child_terminated
2021-04-12 18:45:13.337 [error] <0.168.0> CRASH REPORT Process memsup with 0 neighbours exited with reason: {port_died,normal} in gen_server:handle_common_reply/8 line 751
2021-04-12 18:45:13.348 [error] <0.233.0> CRASH REPORT Process <0.233.0> with 0 neighbours exited with reason: {{port_died,normal},{gen_server,call,[memsup,get_system_memory_data,infinity]}} in gen_server:call/3 line 223
2021-04-12 18:45:13.380 [error] <0.166.0> Supervisor os_mon_sup had child memsup started with memsup:start_link() at <0.168.0> exit with reason {port_died,normal} in context child_terminated
2021-04-12 18:45:13.414 [error] <0.232.0>@vmq_generic_msg_store:-setup_index/1-fun-0-:362 gen_server <0.232.0> terminated with reason: bad argument in call to ets:insert(vmq_generic_msg_store_init_msg_idx_7, {{init,{[],<<"cshea-VirtualBox">>},{1608,598809,71211},<<114,144,28,211,154,59,138,51,46,118,139,...>>}}) in vmq_generic_msg_store:'-setup_index/1-fun-0-'/4 line 362
2021-04-12 18:45:13.421 [error] <0.219.0>@vmq_generic_msg_store_sup:-start_link/0-lc$^0/1-0-:40 CRASH REPORT Process <0.219.0> with 0 neighbours exited with reason: no match of right hand value {error,{{{port_died,normal},{gen_server,call,[memsup,get_system_memory_data,infinity]}},{child,undefined,{vmq_generic_msg_store_bucket,8},{vmq_generic_msg_store,start_link,"\b"},permanent,5000,worker,[vmq_generic_msg_store]}}} in vmq_generic_msg_store_sup:'-start_link/0-lc$^0/1-0-'/2 line 40 in application_master:init/4 line 138
2021-04-12 18:45:13.423 [error] <0.232.0>@vmq_generic_msg_store:-setup_index/1-fun-0-:362 CRASH REPORT Process <0.232.0> with 0 neighbours crashed with reason: bad argument in call to ets:insert(vmq_generic_msg_store_init_msg_idx_7, {{init,{[],<<"cshea-VirtualBox">>},{1608,598809,71211},<<114,144,28,211,154,59,138,51,46,118,139,...>>}}) in vmq_generic_msg_store:'-setup_index/1-fun-0-'/4 line 362
2021-04-12 18:45:13.446 [error] <0.221.0>@vmq_generic_msg_store:-setup_index/1-fun-0-:362 Supervisor vmq_generic_msg_store_sup had child {vmq_generic_msg_store_bucket,7} started with vmq_generic_msg_store:start_link(7) at <0.232.0> exit with reason bad argument in call to ets:insert(vmq_generic_msg_store_init_msg_idx_7, {{init,{[],<<"cshea-VirtualBox">>},{1608,598809,71211},<<114,144,28,211,154,59,138,51,46,118,139,...>>}}) in vmq_generic_msg_store:'-setup_index/1-fun-0-'/4 line 362 in context shutdown_error

Thanks for helping.

**Comment by ioolkos** [comment]
@conni-mpd it looks like VerneMQ is trying to boot the message store plugin (which loads and indexes messages from disk) and then again runs out of disk (and by that failing to boot, as the message store doesn't boot properly).

How much disk space is remaining at this point?

**Comment by conni-mpd** [comment]
@ioolkos Thanks for the quick response. There is still 16.8GB available on the disk.

**Comment by ioolkos** [comment]
@conni-mpd 
There's 2 processes, disksup and memsup, that are supervising OS resources from within VerneMQ. 16GB disk looks enough to me (but I have no details on your use case). So the cause might be memory pressure, while starting and indexing messages from disk.

**Comment by conni-mpd** [comment]
@ioolkos 
Thanks for providing the details. It helped a lot to identify the errors. The disk was once out of space so VerneMQ stopped working. However, such error persisted even after I cleaned up the disk.  I had to manually remove the files (K01vernemq) located in etc/rc0.d and reinstalled the program to resolve the issue.  I am still not sure if this is a bug for verneMQ or Ubuntu running on virtualbox.  

Another thing is that I had verneMQ running on two of my machines (both Ubuntu 18.04 but on different VM host) and both ran out disk space at the same time.  This error only happened on one machine but not the other.
